Sources/TensorFlow/Operators/NN.swift

```diff
@@ -293,279 +406,148 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
-                                    Int32(strides.2), Int32(strides.3)]),
-            padding: padding.raw)
-    }
+/// Computes a 2-D max pooling, with the specified filter sizes, strides, and
+/// padding.
+///
+/// - Parameters:
+///     - filterSize: The dimensions of the pooling kernel.
+///     - strides: The strides of the sliding filter for each dimension of the
+///         input.
+///     - padding: The padding for the operation.
+@differentiable(wrt: x, vjp: _vjpMaxPool2D(filterSize:strides:padding:))
+func maxPool2D<Scalar: TensorFlowFloatingPoint>(
+    filterSize: (Int, Int, Int, Int),
+    strides: (Int, Int, Int, Int),
+    padding: Padding
+    ) -> Tensor<Scalar> {
+    return Raw.maxPoolV2(
+        x,
+        filterSize: Tensor<Scalar><Int32>([Int32(filterSize.0), Int32(filterSize.1),
+                              Int32(filterSize.2), Int32(filterSize.3)]),
+        strides: Tensor<Scalar><Int32>([Int32(strides.0), Int32(strides.1),
+                                Int32(strides.2), Int32(strides.3)]),
+        padding: padding.raw)
+}
 
-    /// Computes a 3-D max pooling, with the specified kernel sizes, strides, and
-    /// padding.
-    ///
-    /// - Parameters:
-    ///     - kernelSize: The dimensions of the pooling kernel.
-    ///     - strides: The strides of the sliding filter for each dimension of the
-    ///         input.
-    ///     - padding: The padding for the operation.
-    @inlinable @inline(__always)
-    @differentiable(
-        wrt: self, vjp: _vjpMaxPooled3D(kernelSize:strides:padding:)
-        where Scalar : TensorFlowFloatingPoint
-    )
-    func maxPooled3D(
-        kernelSize: (Int, Int, Int, Int, Int),
-        strides: (Int, Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.maxPool3D(
-            self,
-            ksize: [Int32(kernelSize.0), Int32(kernelSize.1),
-                    Int32(kernelSize.2), Int32(kernelSize.3), Int32(kernelSize.4)],
-            strides: [Int32(strides.0), Int32(strides.1),
-                      Int32(strides.2), Int32(strides.3), Int32(strides.4)],
-            padding: padding.raw)
-    }
+/// Computes a 3-D max pooling, with the specified filter sizes, strides, and
+/// padding.
+///
+/// - Parameters:
+///     - filterSize: The dimensions of the pooling kernel.
+///     - strides: The strides of the sliding filter for each dimension of the
+///         input.
+///     - padding: The padding for the operation.
+@differentiable(wrt: x, vjp: _vjpMaxPool3D(filterSize:strides:padding:))
+func maxPool3D<Scalar: TensorFlowFloatingPoint>(
+    filterSize: (Int, Int, Int, Int, Int),
+    strides: (Int, Int, Int, Int, Int),
+    padding: Padding
+    ) -> Tensor<Scalar> {
+    return Raw.maxPool3D(
+        self,
+        filterSize: [Int32(filterSize.0), Int32(filterSize.1),
+                Int32(filterSize.2), Int32(filterSize.3), Int32(filterSize.4)],
```

Comment:
Indent to match the first element in this array literal.
```suggestion
                     Int32(filterSize.2), Int32(filterSize.3), Int32(filterSize.4)],
```